src/coreclr/jit/fgflow.cpp

```diff
@@ -235,6 +235,20 @@ FlowEdge* Compiler::fgAddRefPred(BasicBlock* block, BasicBlock* blockPred, FlowE
     //
     assert(block->checkPredListOrder());
 
+    // When initializing preds, ensure edge likelihood is set,
+    // such that this edge is as likely as any other successor edge
+    //
+    if (initializingPreds)
+    {
+        // Shouldn't be copying edge likelihoods/weights from an existing edge
+        //
+        assert(oldEdge == nullptr);
+
+        const unsigned numSucc = blockPred->NumSucc();
+        assert(numSucc > 0);
+        flow->setLikelihood(1.0 / numSucc);
```

Comment:
@AndyAyersMS you mentioned the likelihood calculation should consider the edge's duplicate count. I ran into issues with `BBJ_COND` blocks with true/false branches to the same target, as `NumSucc` handles this case by returning 1 for such blocks. In such cases, if we calculate the likelihood as `(1.0 / numSucc) / dupCount`, the edge likelihood overflows to `(1.0 / 1) * 2 = 2.0`. So leaving the dup count out of the calculation seems to work for that case, but I think this will give us wonky likelihoods for `BBJ_SWITCH` blocks with duplicate successors.

Is it possible to encounter `BBJ_SWITCH` blocks with duplicate successors this early on in the JIT? If so, we can use `BasicBlock::NumSucc(Compiler *)` to get the number of unique switch successors to get the edge likelihoods right, though it'll probably cost a bit more TP.

Reply:
For switches having dups is pretty common.

Maybe you just need to special-case `BBJ_COND` and check for the odd cases where both targets are the same block.